tests/spectral_filtering_test.py

```diff
@@ -0,0 +1,508 @@
+    assert out_wire[2][1] == grid_1_shape
+
+    in_wire = (grid_0_flattened + grid_1_flattened, length_list, [grid_0_shape, grid_1_shape])
+    out_wire = reshape_task(in_wire)
+
+    assert (out_wire[0][0] == grid_0).all()
+    assert (out_wire[0][1] == grid_1).all()
+
+def test_gen_ndarray_to_dict_task():
+
+    SAMPLE_RATE = 20
+    DURATION = 2
+    FREQUENCY = 2
+    _, sine_wave = generate_sine_wave(FREQUENCY, SAMPLE_RATE, DURATION)
+
+    result_list_one = [[[QubitPauliOperator({qps_one: coef * amp}) for coef in sine_wave] for amp in sine_wave] for _ in sine_wave]
+    result_grid_one = np.array(result_list_one)
+    result_dict_one = {
+        qps_one:np.array([[[coef * amp for coef in sine_wave] for amp in sine_wave] for _ in sine_wave])
+    }
+
+    result_list_two = [[QubitPauliOperator({qps_one: coef * amp , qps_two: coef}) for coef in sine_wave] for amp in sine_wave]
+    result_grid_two = np.array(result_list_two)
+    result_dict_two = {
+        qps_one: np.array([[coef * amp for coef in sine_wave] for amp in sine_wave]),
+        qps_two: [[coef for coef in sine_wave] for amp in sine_wave]
+    }
+
+    result_grid_list = [result_grid_one, result_grid_two]
+
+    ndarray_to_dict_task = gen_ndarray_to_dict_task()
+    in_wires = (result_grid_list, )
+    out_wires = ndarray_to_dict_task(in_wires)
+
+    assert (out_wires[0][0][qps_one] == result_dict_one[qps_one]).all()
+    assert (out_wires[0][1][qps_one] == result_dict_two[qps_one]).all()
+    assert (out_wires[0][1][qps_two] == result_dict_two[qps_two]).all()
+
+def test_gen_fft_task():
+
+    SAMPLE_RATE = 20
+    DURATION = 2
+    FREQUENCY = 2
+    _, sine_wave = generate_sine_wave(FREQUENCY, SAMPLE_RATE, DURATION)
+
+    result_dict_one = {
+        qps_one:np.array([[[coef * amp for coef in sine_wave] for amp in sine_wave] for _ in sine_wave])
+    }
+
+    result_dict_two = {
+        qps_one: np.array([[coef * amp for coef in sine_wave] for amp in sine_wave]),
+        qps_two: [[coef for coef in sine_wave] for amp in sine_wave]
+    }
+
+    result_dict_list = [result_dict_one, result_dict_two]
+
+    fft_task = gen_fft_task()
+    in_wires = (result_dict_list, )
+    out_wires = fft_task(in_wires)
+
+    N = SAMPLE_RATE * DURATION
+
+    yf = out_wires[0]
+
+    assert math.isclose(abs(yf[0][qps_one][0][4][4]), N**3/4)
+    assert math.isclose(abs(yf[0][qps_one][0][-4][4]), N**3/4)
+    assert math.isclose(abs(yf[0][qps_one][0][4][-4]), N**3/4)
+    assert math.isclose(abs(yf[0][qps_one][0][-4][-4]), N**3/4)
+
+    assert math.isclose(abs(yf[1][qps_one][4][4]), N**2/4)
+    assert math.isclose(abs(yf[1][qps_one][-4][4]), N**2/4)
+    assert math.isclose(abs(yf[1][qps_one][4][-4]), N**2/4)
+    assert math.isclose(abs(yf[1][qps_one][-4][-4]), N**2/4)
+
+    assert math.isclose(abs(yf[1][qps_two][0][4]), N**2/2)
+    assert math.isclose(abs(yf[1][qps_two][0][-4]), N**2/2)
+
+def test_gen_fft_task_with_sine():
+    
+    SAMPLE_RATE = 20
```

Comment:
why are these variables in full caps? stylistically I think they should be lowercase